src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -341,6 +341,35 @@ public override FormulaValue Visit(RecordFieldAccessNode node, SymbolContext con
                 return left;
             }
 
+            if (left is CustomObjectValue cov)
+            {
+                if (cov.Impl.IsObject)
+                {
+                    if (cov.Impl.TryGetProperty(node.Field.Value, out var res))
+                    {
+                        return new CustomObjectValue(node.IRContext, res);
+                    }
+                    else
+                    {
+                        return new ErrorValue(node.IRContext, new ExpressionError()
+                        {
+                            Message = $"The object does not contain a field with the name {node.Field.Value}",
+                            Span = node.IRContext.SourceContext,
+                            Kind = ErrorKind.BadLanguageCode
+                        });
+                    }
+                }
+                else
+                {
+                    return new ErrorValue(node.IRContext, new ExpressionError()
+                    {
+                        Message = "The CustomObject does not represent an object",
```

Comment:
This works, or maybe something like "Accessing a field is not valid on this value"

Reply:
Changed this message